design/capm3-remediation-controller-proposal.md

```diff
@@ -143,15 +148,16 @@ metadata:
   namespace: NAMESPACE_OF_UNHEALTHY_MACHINE
   ownerReferences:
    - apiVersion: cluster.x-k8s.io/v1alphaX
-     kind: MachineHealthCheck
-     name: REMEDIATION_GROUP
+     kind: Machine
```

Comment:
I'm wondering if we should should add two ownerRefs. One for `Machine` and one for `MachineHealthCheck`.

If we don't add ownerRef to `MachineHealthCheck`, the following scenario may occur:

1. Machine goes unhealthy
2. MHC creates a `Metal3Remediation` CR
3. User deletes corresponding `MachineHealthCheck` CR
4. RC remediates the machine, but MHC won't delete the `Metal3Remediation` CR, since there's no `MachineHealthCheck` associated with the machine. RC will think that the remediation failed, although it succeeded, and will delete the machine eventually.

Reply:
I would use Machine OwnerRef only. What happens if the RC sets a condition and the `Machine` object gets deleted, will the Remediation request be deleted if it still has a `MachineHealthCheck `OwnerRef?